pkg/volume/rbd/rbd_util.go

```diff
@@ -37,19 +38,51 @@ import (
 	"k8s.io/kubernetes/pkg/volume"
 )
 
+// search /sys/bus for rbd device that matches given pool and image
+func getDevFromImageAndPool(pool, image string) string {
+	// /sys/bus/rbd/devices/X/name and /sys/bus/rbd/devices/X/pool
+	sys_path := "/sys/bus/rbd/devices"
+	if dirs, err := ioutil.ReadDir(sys_path); err == nil {
+		for _, f := range dirs {
+			name := f.Name()
```

Comment:
that's cool, is there an efficient way to regex match two string patterns in two filename patterns?